### PR TITLE
14.0 fix sale recurring invoice

### PR DESCRIPTION
**Pull request by mourad-ehm**

### CODE REVIEW

File: fieldservice_account/models/fsm_stage.py [hunk 1]
```diff
@@ -8,3 +8,11 @@ class FSMStage(models.Model):
     _inherit = "fsm.stage"
 
     is_invoiceable = fields.Boolean(copy=False)
+
+    def _get_invoiceable_stage(self):
+        """
+        overide this method to define invoiceable stage
+        by other critaria
```
Comment (hparfr) [inline fieldservice_account/models/fsm_stage.py:15]:
```suggestion
        override this method to define invoiceable stage
        by other criteria
```

File: fieldservice_sale/models/sale_order_line.py [hunk 2]
```diff
@@ -140,12 +140,29 @@ def _field_service_generation(self):
             if rec.product_id.field_service_tracking == "line":
                 rec._field_find_fsm_order()
 
+    def _get_invoiceable_order_domain(self):
```
Comment (hparfr) [inline fieldservice_sale/models/sale_order_line.py:143]:
```suggestion
    def _get_invoiceable_fsm_order_domain(self):
```

File: fieldservice_sale/models/sale_order_line.py [hunk 3]
```diff
@@ -140,12 +140,29 @@ def _field_service_generation(self):
...
+        """
+        overide this method to define more search cretaria for invoiceable
```
Comment (hparfr) [inline fieldservice_sale/models/sale_order_line.py:145]:
```suggestion
        override this method to define more search criteria for invoiceable
```

File: fieldservice_account/models/fsm_stage.py [hunk 4]
```diff
@@ -8,3 +8,11 @@ class FSMStage(models.Model):
...
+        :return:
+        """
+        return self.search([("is_invoiceable", "=", True)])
```
Comment (hparfr) [inline fieldservice_account/models/fsm_stage.py:18]: why not return the domain here instead of the record set ?

**Review by hparfr — CHANGES_REQUESTED** [review]

**Comment by max3903** [comment]
@mourad-ehm Can you split your pull request by module please?

**Pull request closed without merging**